iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/ReplaceTeePointByVoltageLevelOnLine.java

```diff
@@ -271,4 +235,75 @@ public void apply(Network network, boolean throwException,
     }
+
+    private boolean createTopology(LineAdder newLine1Adder, LineAdder newLine2Adder, VoltageLevel tappedVoltageLevel, Reporter reporter, boolean throwException) {
+        TopologyKind topologyKind = tappedVoltageLevel.getTopologyKind();
+        if (topologyKind == TopologyKind.BUS_BREAKER) {
+            Bus bus = tappedVoltageLevel.getBusBreakerView().getBus(bbsOrBusId);
+            if (bus == null) {
+                return errorWhenBusNull(reporter, tappedVoltageLevel, throwException);
+            }
+            Bus bus1 = tappedVoltageLevel.getBusBreakerView()
+                .newBus()
+                .setId(newLine1Id + "_BUS_1")
+                .add();
+            Bus bus2 = tappedVoltageLevel.getBusBreakerView()
+                .newBus()
+                .setId(newLine2Id + "_BUS_2")
+                .add();
+            createBusBreakerSwitches(bus1.getId(), bus.getId(), bus2.getId(), bbsOrBusId, tappedVoltageLevel.getBusBreakerView());
+            newLine1Adder.setBus2(bus1.getId());
+            newLine1Adder.setConnectableBus2(bus1.getId());
+            newLine2Adder.setBus1(bus2.getId());
+            newLine2Adder.setConnectableBus1(bus2.getId());
+        } else if (topologyKind == TopologyKind.NODE_BREAKER) {
+            BusbarSection bbs = tappedVoltageLevel.getNodeBreakerView().getBusbarSection(bbsOrBusId);
+            if (bbs == null) {
+                return errorWhenBusbarSectionNull(reporter, tappedVoltageLevel, throwException);
+            }
+            // New node
+            int firstAvailableNode = tappedVoltageLevel.getNodeBreakerView().getMaximumNodeIndex() + 1;
+            newLine1Adder.setNode2(firstAvailableNode);
+            newLine2Adder.setNode1(firstAvailableNode + 3);
+
+            // Busbar section properties
+            BusbarSectionPosition position = bbs.getExtension(BusbarSectionPosition.class);
+
+            // Topology creation
+            if (position == null) {
+                // No position extension is present so only one disconnector is needed
+                createNodeBreakerSwitchesTopologyFromBusbarSectionList(tappedVoltageLevel, firstAvailableNode, firstAvailableNode + 1, newLine1Id, bbs);
+                createNodeBreakerSwitchesTopologyFromBusbarSectionList(tappedVoltageLevel, firstAvailableNode + 3, firstAvailableNode + 2, newLine2Id, bbs);
+                LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs.getId());
+                noBusbarSectionPositionExtensionReport(reporter, bbs);
+            } else {
+                List<BusbarSection> bbsList = tappedVoltageLevel.getNodeBreakerView().getBusbarSectionStream()
+                    .filter(b -> b.getExtension(BusbarSectionPosition.class) != null)
+                    .filter(b -> b.getExtension(BusbarSectionPosition.class).getSectionIndex() == position.getSectionIndex()).collect(Collectors.toList());
```

Comment:
We're doing this quite often, I think 5 times. We should put this in a method of `TopologyModificationUtils` that would be clearer.

Reply:
getBusbarSectionsFromPosition created

Reply:
`getParallelBusbarSections`? I think you can remove the fromPosition, you don't need to specify the arguments in the name of the method

Reply:
changed